### PR TITLE
Drop support for Python 2.7 and 3.5.

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Comment by dataflake** [comment]
`.travis.yml` and `setup.py` still refer to Python 2

File: setup.py
Comment (dataflake) [inline setup.py]: Can go away

File: setup.py
Comment (dataflake) [inline setup.py]: Can change to `'>=3.6'`

File: setup.py
Comment (dataflake) [inline setup.py]: Can probably go

File: .travis.yml
Comment (mgedmin) [inline .travis.yml:3]: If you're dropping 2.7, please mention that in the PR title.

(Also, I thought there were a few people who wanted to keep maintaining 2.7 support for ZF packages, and the consensus was we weren't going to make their life harder without a good reason?)

File: src/AuthEncoding/AuthEncoding.py
Comment (mgedmin) [inline src/AuthEncoding/AuthEncoding.py]: Why not an f-string if we're going hard for >= 3.6?

File: src/AuthEncoding/AuthEncoding.py
Comment (mgedmin) [inline src/AuthEncoding/AuthEncoding.py]: I 
```suggestion
    for x, y in zip(val1, val2):
```

File: src/AuthEncoding/AuthEncoding.py
Comment (mgedmin) [inline src/AuthEncoding/AuthEncoding.py:126]: I'd replace this whole method with `return os.urandom(7)` (in a separate PR).

**Review by mgedmin — APPROVED** [review]
lgtm, assuming we're fine with dropping 2.7 as a group

(I wanted to link to a previous discussion about that but couldn't find it)

**Comment by dataflake** [comment]
> lgtm, assuming we're fine with dropping 2.7 as a group

Michael worked on several packages yesterday and wanted to just drop 3.5 first, because Zope 5 doesn't support it. But he overlooked that dropping 3.5 would make a package potentially incompatible with Zope 4.x, which explicitly supports it. Then we both reasoned that he could just make the full step of dropping 2.7 as well, so the package would end up for Zope 5 only. But Jason Madden spoke up about some packages where he definitely wants to retain 2.7.

I'm now thinking that we should stop such efforts to drop compatibility unless there are really good reasons and revisit it when Zope 5 has been around a bit and adopted by the bigger "customers" in the ecosystem, e.g. Plone.

**Comment by dataflake** [comment]
To get this out of the critical path for a Zope 5 release I have pulled out the changes that add support for Python 3.8 and Python 3.9 and merged those into `master`. I then published release 4.2, which we can use for Zope 5 (and 4).

This PR only contains the changes for dropping Python 2 and Python 3.5 support, and that should wait until we have consensus on when to drop this support for packages where keeping that support doesn't introduce a big burden. I added the `do not merge` tag for that reason.

**Comment by dataflake** [comment]
Superceded by #14

**Pull request closed without merging**
